Sources/SemanticAnalyzer/SemanticAnalyzer+Statements.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  SemanticAnalyzer+Statements.swift
+//  SemanticAnalyzer
+//
+//  Created by Farcas, Calin on 10/11/2018.
+//
+import Foundation
+import AST
+import Lexer
+import Diagnostic
+
+extension SemanticAnalyzer {
+  public func postProcess(ifStatement: IfStatement,
+                          passContext: ASTPassContext) -> ASTPassResult<IfStatement> {
+    let ifStatement = ifStatement
+    let condition = ifStatement.condition
+    let environment = passContext.environment!
+    let enclosingTypeIdentifier = passContext.enclosingTypeIdentifier!
+    let typeStates = passContext.contractBehaviorDeclarationContext?.typeStates ?? []
+    let callerProtections = passContext.contractBehaviorDeclarationContext?.callerProtections ?? []
+    var diagnostics = [Diagnostic]()
+
+    // Allow let statements
```

Comment:
Is it possible to check that we are trying to assign a variable to an optional result? Swift allows this
`if let x: Int = 2 {
    print("Programmer may think x is optional.")
}`
but I don't see a good reason why.

Reply:
I think it's a reasonable suggestion, but a bit out of the scope of this ticket.